### ISSUE TITLE
KMeans Learn 

### ISSUE THREAD

**Issue by petrasvestartas** [accord-net/framework#360]
Hi,

How can I retrieve an array of indices the same as using method KMeans.Compute, but using KMeans.Learn?

I cannot find any example using KMeans.Learn

Thanks

**Comment by cesarsouza** [comment]
Hi there,

You can use the Decide method of the generated cluster collection. For example:

```
             Accord.Math.Random.Generator.Seed = 0;

            // Declare some observations
            double[][] observations =
            {
                new double[] { -5, -2, -1 },
                new double[] { -5, -5, -6 },
                new double[] {  2,  1,  1 },
                new double[] {  1,  1,  2 },
                new double[] {  1,  2,  2 },
                new double[] {  3,  1,  2 },
                new double[] { 11,  5,  4 },
                new double[] { 15,  5,  6 },
                new double[] { 10,  5,  6 },
            };

            double[][] orig = observations.MemberwiseClone();

            // Create a new K-Means algorithm with 3 clusters 
            KMeans kmeans = new KMeans(3);

            // Compute the algorithm, retrieving an integer array
            //  containing the labels for each of the observations
            KMeansClusterCollection clusters = kmeans.Learn(observations);

            // As a result, the first two observations should belong to the
            //  same cluster (thus having the same label). The same should
            //  happen to the next four observations and to the last three.
            int[] labels = clusters.Decide(observations);
```

The documentation is still being updated, sorry about that.

Regards,
Cesar

**Comment by petrasvestartas** [comment]
Thank you very much for an answer:)

**Comment by petrasvestartas** [comment]
Does the kmeans works with negative values?

**Comment by cesarsouza** [comment]
Well, yes, it definitely should. But if you think you have found a problem, please let me know!

**Comment by y-takashina** [comment]
Added @cesarsouza's sample code to https://github.com/accord-net/framework/wiki/Clustering

**Comment by cesarsouza** [comment]
Fixed in 3.5.0.

@y-takashina many thanks for the sample code addition!
